src/err/mod.rs

```diff
@@ -390,6 +390,29 @@ impl PyErr {
         PyErr::from_state(PyErrState::Normalized(self.normalized(py).clone()))
     }
 
+    /// Return the cause (either an exception instance, or None, set by `raise ... from ...`)
+    /// associated with the exception, as accessible from Python through `__cause__`.
+    pub fn cause(&self, py: Python) -> Option<PyErr> {
+        let ptr = unsafe { ffi::PyException_GetCause(self.pvalue(py).as_ptr()) };
+        let obj = unsafe { py.from_owned_ptr_or_opt::<PyAny>(ptr) };
+        obj.map(|x| Self::from_instance(x))
+    }
+
+    /// Set the cause associated with the exception
+    pub fn set_cause(&self, py: Python, cause: Self) {
+        let cause = cause.into_instance(py);
+        unsafe {
+            ffi::PyException_SetCause(self.pvalue(py).as_ptr(), cause.as_ptr());
+        }
+    }
```

Comment:
What do you think about `set_cause(&self, py: Python, cause: Option<Self>)` instead? Then there's no need for `clear_cause`

It has a nice symmetry with `get_cause() -> Option<Self>`, but I guess it will force users to type `Some()` a lot.

Reply:
👍